server/ai_mediaserver.go

```diff
@@ -370,17 +371,26 @@ func (ls *LivepeerServer) StartLiveVideo() http.Handler {
 			return
 		}
 		requestID := string(core.RandomManifestID())
+		ctx := clog.AddVal(r.Context(), "request_id", requestID)
+		clog.Infof(ctx, "Received live video AI request for %s", streamName)
+
+		// Kick off the RTMP pull and segmentation as soon as possible
+		ssr := media.NewSwitchableSegmentReader()
+		go func() {
+			media.RunSegmentation("rtmp://localhost/"+streamName, ssr.Read)
```

Comment:
Does it mean, that Gateway will pull the stream from MediaMTX? Isn't it opposite to what you designed on the diagram?

<img width="635" alt="image" src="https://github.com/user-attachments/assets/c1e584e2-b1c4-4a43-8adc-07fe29d74688">

Reply:
Yes, gateway initiates the RTMP pull, but the flow of media still goes from mediamtx -> gateway

I guess there could be a better distinction between "who initiates the pull" vis-a-vis the actual flow of media, but RTMP isn't really a request-response protocol in the same way the other HTTP flows are in this diagram. Open to suggestions.